apolo-cli/src/apolo_cli/formatters/apps.py

```diff
@@ -41,3 +44,103 @@ def __call__(self, apps: List[App]) -> Table:
                 app.state,
             )
         return table
+
+
+class BaseAppEventsFormatter:
+    def __call__(self, events: List[AppEvent]) -> Union[Table, Group, str]:
```

Comment:
We've dropped Python 3.9 support, please feel free to use `Table | Group | str` instead.

Also, do you really need an enumeration type? Maybe `rich.console.RenderableType` fits better?

Plus, `list[AppEvent]` could be used instead of `List[AppEvent']`

The same is applicable for the rest of the PR